docs/mutations.md

```diff
@@ -269,13 +348,127 @@ print(f"Before: {mts.num_mutations} mutations.\n"
       f"After: {mmts.num_mutations} mutations.")
 ```
 
-Adding ancestral mutations can result in confusing situations, so should be
-done with care. For instance, new mutations added to an already mutated site
+This might be done, for instance, if a tree sequence is the result of a forwards
+simulation that has non-neutral variation, and you'd like to add neutral mutations.
+
+(sec_mutations_state_independence)=
+
+### Silent mutations and state-independence
+
+However, adding mutations in a way that might result in a new mutation being
```

Comment:
A minor nit here: I'm imagining each of these sections as being something you might land into from a completely different part of the documentation by linking in, and not necessarily as part of a linear narrative. Starting with "However" like this is jarring from that perspective, implying that you need to have read all the preceeding stuff to be able to follow what's going on.